dialect/entsql/annotation.go

```diff
@@ -202,6 +202,28 @@ type IndexAnnotation struct {
 	//	CREATE INDEX `table_c1_c2_c3` ON `table`(`c1`(100), `c2`(200), `c3`)
 	//
 	PrefixColumns map[string]uint
+
+	// Desc defines the DESC clause for a single column index.
+	// In MySQL, the following annotation maps to:
+	//
+	//	index.Fields("column").
+	//		Annotation(entsql.Desc())
+	//
+	//	CREATE INDEX `table_column` ON `table`(`column` DESC)
+	//
+	Desc bool
+
+	// DescColumns defines the DESC clause for columns in a multi column index.
+	// In MySQL, the following annotation maps to:
+	//
+	//	index.Fields("c1", "c2", "c3").
+	//		Annotation(
+	//			entsql.DescColumns("c1", "c2"),
+	//		)
+	//
+	//	CREATE INDEX `table_c1_c2_c3` ON `table`(`c1` DESC, `c2` DESC, `c3`)
+	//
+	DescColumns map[string]bool
```

Comment:
The more "goish" (and in my opinion cleaner way) is going with `map[string]struct{}`. Not really sure though, the check adds some boilerplate to it. 

Will this be used by the user directly? Looks like it is unlikely.

Reply:
You still have the option to add annotations like:
```go
&entsql.IndexAnnotation{
	Prefix: 10,
	Desc: true,
}
```